src/components/ImageBox.js

```diff
@@ -4,21 +4,25 @@ import { PokemonImage } from "./PokemonImage";
 
 export const ImageBox = ({randomPokemons}) => {
   const [imageBox, setImageBox] = useState([]);
+
+  const addImageToBox = useCallback(
+    (id) => {
+      console.log(randomPokemons);
+      const imageToAdd = randomPokemons.filter((pokemon) => pokemon.id === id);
+      console.log(imageToAdd);
+      setImageBox((prev) => [...prev, imageToAdd[0]]);
+      console.log(imageBox);
+    },
+    [imageBox, randomPokemons]
+  );
+
   const [{ isOver }, drop] = useDrop(() => ({
     accept: "image",
     drop: (item) => (addImageToBox(item.id)),
     collect: (monitor) => ({
       isOver: !!monitor.isOver(),
     }),
-  }));
-
-  const addImageToBox = useCallback((id) => {
-    console.log(randomPokemons);
-    const imageToAdd = randomPokemons.filter((pokemon) => pokemon.id === id);
-    console.log(imageToAdd);
-    setImageBox(prev => [...prev, imageToAdd[0]]);
-    console.log(imageBox);
-  }, [imageBox, randomPokemons]);
+  }), [randomPokemons]);
```

Comment:
This is the fix. Adding the `randomPokemons` prop as a dependency of the `useDrop` hook. Now whenever `addImageToBox` is called, it will have access to this variable.